### ISSUE TITLE
Use Snapshot Testing for testing model inference

### ISSUE THREAD

**Issue by nordzilla** [mozilla/translations#946]
We now have several test files in [inference/wasm/tests/test-cases](https://github.com/mozilla/translations/tree/main/inference/wasm/tests/test-cases) that compare before-and-after translations. 

It would be a good idea to transition these test cases to use vitest's [Snapshot](https://vitest.dev/guide/snapshot) capabilities, so that it is easy to change expected assertions when adding new models, or if models change, etc.

**Comment by gregtatum** [comment]
My preference from other projects is to use inline snapshots if possible. It makes it easier to review and maintain tests. They are a bit more magical though.

**Comment by nordzilla** [comment]
I would say that inline snapshots via data-driven test assertions is effectively what I've already implemented. 

Does that mean that you're overall fine with the current format?

**Comment by gregtatum** [comment]
Yeah I'm fine with how things are implemented, but it won't work with the `matchInlineSnapshot` feature, as that rewrites the files in place:

```
import { expect, it } from 'vitest'

it('toUpperCase', () => {
  const result = toUpperCase('foobar')
  expect(result).toMatchInlineSnapshot('"FOOBAR"')
})
```

The benefits of snapshot testing is that the tests can be updated through an automated command as opposed to manually updating the examples yourself. I have found non-inline snapshot hard to use in the past as the expectations are separated from the actual results.
